### PR TITLE
[Serve] Promote autoscaling feature

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?

1. get rid of the private attribute
2. fix unit test
3. docs and workflows

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by architkulkarni — COMMENTED** [review]
There should be some minor corresponding changes to make in the docs right?  I think it makes sense to have those in this PR.

**Comment by sihanwang41** [comment]
> There should be some minor corresponding changes to make in the docs right? I think it makes sense to have those in this PR.

good point, updated!

File: doc/source/serve/managing-deployments.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ and leave us feedback through [Github Issues](https://github.com/ray-project/ray
```
Comment (architkulkarni) [inline doc/source/serve/managing-deployments.md:87]: Could you also remove "experimental" from the above?  If it's  still "beta", we can leave the warning about it being subject to change, as long as we remove the word "experimental"

File: doc/source/serve/managing-deployments.md [hunk 2]
```diff
@@ -76,18 +76,14 @@ func.options(num_replicas=1).deploy()
 
 #### Autoscaling
 
-Serve also has experimental support for a demand-based replica autoscaler.
+Serve also has the support for a demand-based replica autoscaler.
```
Comment (edoakes) [inline doc/source/serve/managing-deployments.md:79]:
```suggestion
Serve also has support for a demand-based replica autoscaler.
```

**Review by edoakes — APPROVED** [review]
💯

File: doc/source/serve/managing-deployments.md [hunk 3]
```diff
@@ -76,18 +76,14 @@ func.options(num_replicas=1).deploy()
...
 It reacts to traffic spikes via observing queue sizes and making scaling decisions.
-To configure it, you can set the `_autoscaling` field in deployment options.
+To configure it, you can set the `autoscaling` field in deployment options.
```
Comment (architkulkarni) [inline doc/source/serve/managing-deployments.md:81]:
```suggestion
To configure it, you can set the `autoscaling_config` field in deployment options.
```

**Comment by simon-mo** [comment]
merge conflict

**Comment by edoakes** [comment]
Tests failing on master.

**Pull request merged**